File: JavaScript/stringcompression.js
let compress = function(chars){
let n=chars.length
let write = 0, read = 0

while(read<n){
    let char = chars[read]
    let count = 0; 
    while(read<n && chars[read]===char){
        read++
        count++
    }
    chars[write++] = char //assigns value of char to chars[write], write is the index position. incrementing write afterwards
    if(count>1){
        for(digit of count.toString()){
            chars[write++] = digit  //now assign the counts value to write position (after increment).
        }
    }

    
}
return write;
}

File: Python/StringCompression.py
class Solution:
    def compress(self, chars: list[str]) -> int:
        write = 0
        read = 0

        while read < len(chars):

            current_char = chars[read]
            count = 0

            while read < len(chars) and current_char == chars[read]:
                read += 1
                count += 1

            chars[write] = current_char
            write += 1

            if count > 1:
                for digit in str(count):
                    chars[write] = digit
                    write += 1

        return write


solution = Solution()
print(solution.compress(["a", "a", "b", "b", "c", "c", "c"]))
